provider/execution/execution.go

```diff
@@ -301,34 +301,42 @@ func (e TableExecutor) resolveResources(ctx context.Context, meta schema.ClientM
 // saveToStorage copies resource data to source, it has ways of inserting, first it tries the most performant CopyFrom if that does work it bulk inserts,
 // finally it inserts each resource separately, appending errors for each failed resource, only successfully inserted resources are returned
 func (e TableExecutor) saveToStorage(ctx context.Context, resources schema.Resources, shouldCascade bool) (schema.Resources, diag.Diagnostics) {
+	var diags diag.Diagnostics
 	if l := len(resources); l > 0 {
 		e.Logger.Debug("storing resources", "count", l)
 	}
 	err := e.Db.CopyFrom(ctx, resources, shouldCascade)
 	if err == nil {
-		return resources, nil
+		return resources, diags
 	}
 	e.Logger.Warn("failed copy-from to db", "error", err)
+	diags = diags.Add(diag.TelemetryFromError(err, diag.CopyFromFailed))
 
 	// fallback insert, copy from sometimes does problems, so we fall back with bulk insert
 	err = e.Db.Insert(ctx, e.Table, resources, shouldCascade)
 	if err == nil {
-		return resources, nil
+		return resources, diags
 	}
 	e.Logger.Error("failed insert to db", "error", err)
+	diags = diags.Add(diag.TelemetryFromError(err, diag.BulkInsertFailed))
 	// Setup diags, adding first diagnostic that bulk insert failed
-	diags := diag.Diagnostics{}.Add(ClassifyError(err, diag.WithType(diag.DATABASE), diag.WithSummary("failed bulk insert on table %q", e.Table.Name)))
+	diags = diags.Add(ClassifyError(err, diag.WithType(diag.DATABASE), diag.WithSummary("failed bulk insert on table %q", e.Table.Name)))
 	// Try to insert resource by resource if partial fetch is enabled and an error occurred
 	partialFetchResources := make(schema.Resources, 0)
+	var failed error
 	for id := range resources {
 		if err := e.Db.Insert(ctx, e.Table, schema.Resources{resources[id]}, shouldCascade); err != nil {
+			failed = err
```

Comment:
would it make sense to also capture the percentage of resources failed? or at least a bool to tell us if it failed for "all resources" or just a few.

Reply:
Added appropriate information, take a look, please.

Reply:
Isn't that information able to be calculated based on the final message?

Reply:
I just added it based on Kemal's comment. :)